Support Decimal DIV changes in cudf [skip ci]

razajafri opened the pull request:
@codereport is making changes to the way `DIV` will behave for fixed-point types #7435. This PR contains Java changes to support those changes. 

Note: This is a draft until #7435 is merged

<!--

Thank you for contributing to cuDF :)

Here are some guidelines to help the review process go smoothly.

1. Please write a description in this text box of the changes that are being
   made.

2. Please ensure that you have written units tests for the changes made/features
   added.

3. There are CI checks in place to enforce that committed code follows our style
   and syntax standards. Please see our contribution guide in `CONTRIBUTING.MD`
   in the project root for more information about the checks we perform and how
   you can run them locally.

4. If you are closing an issue please use one of the automatic closing words as
   noted here: https://help.github.com/articles/closing-issues-using-keywords/

5. If your pull request is not ready for review but you want to make use of the
   continuous integration testing facilities please label it with `[WIP]`.

6. If your pull request is ready to be reviewed without requiring additional
   work on top of it, then remove the `[WIP]` label (if present) and replace
   it with `[REVIEW]`. If assistance is required to complete the functionality,
   for example when the C/C++ code of a feature is complete but Python bindings
   are still required, then add the label `[HELP-REQ]` so that others can triage
   and assist. The additional changes then can be implemented on top of the
   same PR. If the assistance is done by members of the rapidsAI team, then no
   additional actions are required by the creator of the original PR for this,
   otherwise the original author of the PR needs to give permission to the
   person(s) assisting to commit to their personal fork of the project. If that
   doesn't happen then a new PR based on the code of the original PR can be
   opened by the person assisting, which then will be the PR that will be
   merged.

7. Once all work has been done and review has taken place please do not add
   features or make changes out of the scope of those requested by the reviewer
   (doing this just add delays as already reviewed code ends up having to be
   re-reviewed/it is hard to tell what is new etc!). Further, please do not
   rebase your branch on main/force push/rewrite history, doing any of these
   causes the context of any comments made by reviewers to be lost. If
   conflicts occur against main they should be resolved by merging main
   into the branch used for making the pull request.

Many thanks in advance for your cooperation!

-->

razajafri commented:
Now that #7435 is merged. The plugin will stop working unless we merge this

mythrocks commented on java/src/main/java/ai/rapids/cudf/ColumnView.java:
```diff
@@ -71,6 +71,13 @@ public final long getNativeView() {
     return viewHandle;
   }
 
+  public static int getFixedPointOutpuScale(BinaryOp op, DType lhsType, DType rhsType) {
```
Typo: `getFixedPointOutputScale()`.

mythrocks replied:
Also, missing docs.

jlowe commented on java/src/main/java/ai/rapids/cudf/ColumnView.java:
```diff
@@ -71,6 +71,13 @@ public final long getNativeView() {
     return viewHandle;
   }
 
+  public static int getFixedPointOutpuScale(BinaryOp op, DType lhsType, DType rhsType) {
```
If this remains a public function it needs the full Javadoc treatment.

mythrocks reviewed (approved):
A couple of minor nitpicks, but looks good to me.

mythrocks reviewed (changes requested):
(Accidentally approved, earlier.) 
Minor nitpicks: Javadoc + typo in method name.

jlowe commented on java/src/main/native/src/ColumnViewJni.cpp:
```diff
@@ -1026,6 +1027,15 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnView_binaryOpVV(JNIEnv *env, j
 }
 
+JNIEXPORT jint JNICALL Java_ai_rapids_cudf_ColumnView_fixedPointOutputScale(JNIEnv *env, jclass, jint int_op,
+                                                                  jint lhs_scale, jint rhs_scale) {
```
Indentation is off here.

gerashegalov reviewed (commented):
@razajafri can you merge the parent branch-0.19 to consume `binary_operation_fixed_point_scale` into your PR (, and address the previous comments on the PR?

Otherwise, LGTM

razajafri commented:
Addressed comments. PTAL 

@jlowe do you have any more concerns? I added a small change to the CMakeLists.txt for not overriding the `CUDF_CPP_BUILD_DIR` if its set in the environment. This is needed to build the bindings within the compose docker. Let me know if you want me to spin this change into a separate PR.

jlowe commented on java/src/main/native/CMakeLists.txt:
```diff
@@ -54,7 +54,9 @@ message(VERBOSE "CUDF_JNI: Statically link the CUDA runtime: ${CUDA_STATIC_RUNTI
 
 set(CUDF_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../../../cpp")
-set(CUDF_CPP_BUILD_DIR "${CUDF_SOURCE_DIR}/build")
+if (NOT CUDF_CPP_BUILD_DIR) 
```
Was this supposed to be `if (NOT DEFINED CUDF_CPP_BUILD_DIR)`?  Otherwise there's no variable expansion in what is written that I can see.

jlowe replied:
Oh nevermind, this should be OK as-is.

jlowe commented on java/src/main/java/ai/rapids/cudf/ColumnView.java:
```diff
@@ -129,7 +129,7 @@ public final long getNativeView() {
   }
 
-  public static int getFixedPointOutpuScale(BinaryOp op, DType lhsType, DType rhsType) {
+  protected static int getFixedPointOutputScale(BinaryOp op, DType lhsType, DType rhsType) {
```
This should be package-private.

razajafri commented:
@jlowe PTAL.

gerashegalov reviewed (approved):
LGTM

jlowe commented:
@gpucibot merge
